src/Sarif.TestUtilities/FileDiffingTests.cs

```diff
@@ -4,15 +4,93 @@
 using System;
 using System.Globalization;
 using System.IO;
+using System.Reflection;
+using FluentAssertions;
 using Microsoft.CodeAnalysis.Sarif.Readers;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using Xunit.Abstractions;
 
 namespace Microsoft.CodeAnalysis.Sarif
 {
     public abstract class FileDiffingTests
     {
-        protected string GenerateDiffCommand(string expected, string actual)
+        public static string GetTestDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@".\TestData", subdirectory));
+        }
+
+        // Retrieving the source path of the tests is only used in developer ad hoc
+        // rebaselining scenarios. i.e., this path won't be consumed by AppVeyor.
+        public static string GetProductTestDataDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@"..\..\..\..\..\src\Sarif.UnitTests\TestData", subdirectory));
+        }
+
+
+        private ITestOutputHelper _outputHelper;
+
+        public FileDiffingTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+
+            if (Directory.Exists(OutputFolderPath))
+            {
+                Directory.Delete(OutputFolderPath, recursive: true);
+            }
+
+            Directory.CreateDirectory(OutputFolderPath);
+        }
+
+
+        protected virtual Assembly ThisAssembly { get { return this.GetType().Assembly; } }
+
+        protected virtual string TypeUnderTest { get { return this.GetType().Name.Substring(0, this.GetType().Name.Length - "Tests".Length); } }
+
+        protected virtual string OutputFolderPath { get { return Path.Combine(Path.GetDirectoryName(ThisAssembly.Location), "UnitTestOutput." + TypeUnderTest); } }
+
+        protected virtual string ProductTestDataDirectory {  get { return GetProductTestDataDirectory(TypeUnderTest); } }
+
+        protected virtual string TestLogResourceNameRoot { get { return "Microsoft.CodeAnalysis.Sarif.UnitTests.TestData." + TypeUnderTest; } }
+
+        protected string GetOutputFilePath(string directory, string resourceName)
+        {
+            string fileName = string.Format("{0}.sarif",
```

Comment:
FxCop would complain that you didn't specify `CultureInfo.InvariantCulture`.

Reply:
good fix of code i moved. using string.format for what is effectively simple string concat not a good idea, from a micro-optimization standpoint.

---
In reply to: [226788994](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226788994) [](ancestors = 226788994)